### PR TITLE
Regression with DatePicker and checkboxes on stock page (catalogue > stock)

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | DatePicker field not working due to wrong css selector. Wrong usage of v-model in PsCheckbox.vue (https://fr.vuejs.org/v2/guide/forms.html#Checkbox)
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Partial http://forge.prestashop.com/browse/BOOM-5620
| How to test?  | Try to click on datepicker field, for example in product page, Available quantity field.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Same problem as @mickaelandrieu on #9137 
@fatmaBouchekoua @khouloudbelguith Can you check please :)

**Comment by fatmaBouchekoua** [comment]
Hello @PierreRambaud , 
I'm checking it. That's weird because it works perfectly on local.

**Comment by PierreRambaud** [comment]
@fatmaBouchekoua Could you please send me the command you used to run this test? :)

**Comment by PierreRambaud** [comment]
@fatmaBouchekoua I can reproduce the bug, while running E2E tests, I see the green message coming and disappeared one second after, and just after the check to `verify the appearance of the green validation`  is executed.

**Comment by fatmaBouchekoua** [comment]
Hello @PierreRambaud ,
To run this test I use this command : npm test -- --URL=URLFrontOffice
That's weird, this is working for me on local shop and on Travis https://travis-ci.org/fatmaBouchekoua/PrestaShop/jobs/385647686
I will try to reproduce it on this branch even if this PR doen't have an impact on the product page.

**Comment by PierreRambaud** [comment]
@fatmaBouchekoua Hey, I found what happened, a problem with package-lock.json file, it's now fix we will release a patch on 1.7.4.x and develop :)

File: admin-dev/themes/new-theme/js/theme.js [hunk 1]
```diff
@@ -47,6 +42,13 @@ import './product-page/index';
...
 import Header from './header.js';
+import initDatePickers from './app/utils/datepicker';
+
+const $ = global.$;
 
 new NavBar();
 new Header();
+
+$(() => {
+  initDatePickers();
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/theme.js:53]: This probably should be done only in pages containing datepickers, shouldn't it?

**Review by eternoendless — CHANGES_REQUESTED** [review]
Please commit built assets in a separate commit, it will make rebasing easier

File: admin-dev/themes/new-theme/js/theme.js (hunk 1 above)
Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/theme.js:53]: It's a theme feature to have a DatePicker, it's not really specific to a single page.
When this selector isn't found, nothing happened, and it cost maybe 0.0005ms (:trollface:). 
If we decide to move this only in pages containing datepickers, we maybe need to do the same for `product-page/index`, `./clickable-dropdown`, and `./translation-page/index` that are in `theme.js`. Wdyt?

**Comment by PierreRambaud** [comment]
@eternoendless I rebase everything in two separate commits :) Sorry for the delay

File: admin-dev/themes/new-theme/js/theme.js (hunk 1 above)
Comment (mickaelandrieu) [reply admin-dev/themes/new-theme/js/theme.js:53]: We need to check that adding it globally doesn't conflict with datepickers management in product page.

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/theme.js:53]: @mickaelandrieu It's a fix because no datepickers working on product page =)

Comment (eternoendless) [reply admin-dev/themes/new-theme/js/theme.js:53]: Ok but we will have to do it eventually when we'll split the files per page sometime in the future

**Review by eternoendless — DISMISSED** [review]

File: admin-dev/themes/new-theme/js/theme.js [hunk 2]
```diff
@@ -47,6 +42,13 @@ import './product-page/index';
 import './translation-page/index';
 
 import Header from './header.js';
+import initDatePickers from './app/utils/datepicker';
+
+const $ = global.$;
```
Comment (eternoendless) [inline admin-dev/themes/new-theme/js/theme.js:47]: You may need to use `window` instead of `global`

Comment (PierreRambaud) [reply admin-dev/themes/new-theme/js/theme.js:47]: Should I need to replace every occurrences  of `global` in other files?

**Comment by marionf** [comment]
Hello @PierreRambaud 

It's ok for checkbox
It's ok for datepicker in quantities tab for a standard product

It's not ok for datepicker in combination tab
capture du 2018-06-04 09-23-17

It's not ok for datepicker in stock movement page
capture du 2018-06-04 09-24-13

File: admin-dev/themes/new-theme/js/theme.js (hunk 2 above)
Comment (eternoendless) [reply admin-dev/themes/new-theme/js/theme.js:47]: No, it will be automagically converted to `window` by webpack. It's just to avoid linter warnings.

**Comment by eternoendless** [comment]
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE98YN54jL_5HJN51YG:bccm2kn) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
> ```Vue
>       onClick() {
>         this.checked = !this.checked;
>       },
> ```

Doesn't removing this prevent the model from being updated?

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133)*

**Comment by PierreRambaud** [comment]
Review status: 14 of 15 files reviewed at latest revision, 1 unresolved discussion.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE9D0j_1XxIexuuYk_4:bgyajt8) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
<details><summary><i>Previously, eternoendless (Pablo Borowicz) wrote…</i></summary><blockquote>

Doesn't removing this prevent the model from being updated?
</blockquote></details>

No, in fact, model and item aren't used here (only to emit checked event to parent.
And it's against the VueJs philosophy https://fr.vuejs.org/v2/guide/forms.html#Checkbox by setting a v-model on the input, the checked value is already toggled.

---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133)*

**Comment by eternoendless** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue, line 49 at r4](https://reviewable.io/reviews/prestashop/prestashop/9133#-LE98YN49owq8CJaSPND:-LE9ufLG86VZpiKoTojK:bh25lys) ([raw file](https://github.com/prestashop/prestashop/blob/5243d55372fd40ad9f99c09e598f7f1379ce608b/admin-dev/themes/new-theme/js/app/widgets/ps-checkbox.vue#L49)):*
<details><summary><i>Previously, PierreRambaud (GoT) wrote…</i></summary><blockquote>

No, in fact, model and item aren't used here (only to emit checked event to parent.
And it's against the VueJs philosophy https://fr.vuejs.org/v2/guide/forms.html#Checkbox by setting a v-model on the input, the checked value is already toggled.
</blockquote></details>



---


*Comments from [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9133#-:-LE9uw_lCaV-6nv7H_wI:bnfp4nl)*

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
Thank you @PierreRambaud

**Pull request merged**